tasks.py

```diff
@@ -849,3 +908,88 @@ def deps(  # noqa: PLR0913
         cmds.append("-c constraints-dev.txt")
 
     ctx.run(" ".join(cmds), echo=True, pty=True)
+
+
+@invoke.task(
+    iterable=["service_names"],
+)
+def ci_tests(
+    ctx: Context,
+    marker: str,
+    verbose: bool = False,
+    reports: bool = False,
+    up_services: bool = False,
+):
+    """
+    Run tests in CI.
+
+    This method looks up the pytest marker provided and runs the tests for that marker,
+    as well as looking up any required services, testing dependencies and extra CLI flags
+    that are need and starting them if `up_services` is True.
+
+    `up_services` is False by default to avoid starting services which may already be up
+    when running tests locally.
+
+    Defined this as a new invoke task to avoid some of the baggage of our old test setup.
+    """
+    pytest_cmds = [
+        "pytest",
+        "-m",
+        f"'{marker}'",
+        "-rEf",
+    ]
+    if reports:
+        pytest_cmds.extend(["--cov=great_expectations", "--cov-report=xml"])
+
+    if verbose:
+        pytest_cmds.append("-vv")
+
+    for test_deps in _get_marker_dependencies(marker):
+        for extra_pytest_arg in test_deps.exta_pytest_args:
+            pytest_cmds.append(extra_pytest_arg)
+
+        for setup_func in test_deps.setup_funcs:
+            setup_func(ctx)
+
+        if test_deps.services and up_services:
```

Comment:
If we remove the `services` flag we can remove this check and the function below. It looks like we don't use this now?

Reply:
The function below is a task which we use.